packages/datatrak-web/src/features/SurveyResponseModal.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import { useForm, FormProvider } from 'react-hook-form';
+import { useSearchParams } from 'react-router-dom';
+import styled from 'styled-components';
+import { Dialog, Typography } from '@material-ui/core';
+import {
+  ModalContentProvider,
+  ModalFooter,
+  ModalHeader,
+  SpinningLoader,
+} from '@tupaia/ui-components';
+import { useSurveyResponseWithForm } from '../api/queries';
+import { Button, SurveyTickIcon } from '../components';
+import { displayDate } from '../utils';
+import { SurveyReviewSection } from './Survey/Components';
+import { SurveyContext } from '.';
+
+const Header = styled.div`
+  display: flex;
+  align-items: center;
+  padding: 0.5rem;
+  width: 100%;
+
+  .MuiSvgIcon-root {
+    font-size: 2.5em;
+    margin-right: 0.35em;
+  }
+`;
+
+const Heading = styled(Typography).attrs({
+  variant: 'h2',
+})`
+  font-size: 1.5rem;
+  color: ${({ theme }) => theme.palette.text.primary};
+  font-weight: 600;
+  margin-bottom: 0.2rem;
+  ${({ theme }) => theme.breakpoints.down('sm')} {
+    font-size: 1rem;
+  }
+`;
+
+const SubHeading = styled(Typography)`
+  color: ${({ theme }) => theme.palette.text.secondary};
+  font-weight: 400;
+  font-size: 1rem;
+  ${({ theme }) => theme.breakpoints.down('sm')} {
+    font-size: 0.875rem;
+  }
+`;
+
+const Loader = styled(SpinningLoader)`
+  width: 25rem;
+  max-width: 100%;
+`;
+
+const getSubHeadingText = surveyResponse => {
+  if (!surveyResponse) {
+    return null;
+  }
+  const date = displayDate(surveyResponse.dataTime);
+  const country = surveyResponse?.countryName;
```

Comment:
```suggestion
  const country = surveyResponse.countryName;
```
surveyResponse must exist at this point